node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -145,11 +145,16 @@ impl<C: Context> ActiveValidator<C> {
         }
         let mut effects = vec![];
         if self.should_send_confirmation(vhash, now, state) {
-            let panorama = state.confirmation_panorama(self.vidx, vhash);
-            if panorama.has_correct() {
-                let confirmation_unit = self.new_unit(panorama, now, None, state, instance_id, rng);
-                let vv = ValidVertex(Vertex::Unit(confirmation_unit));
-                effects.extend(vec![Effect::NewVertex(vv)])
+            let panorama_opt = state.confirmation_panorama(self.vidx, vhash);
+            if let Some(panorama) = panorama_opt {
+                if panorama.has_correct() {
+                    let confirmation_unit =
+                        self.new_unit(panorama, now, None, state, instance_id, rng);
+                    let vv = ValidVertex(Vertex::Unit(confirmation_unit));
+                    effects.extend(vec![Effect::NewVertex(vv)])
+                }
+            } else {
+                return effects;
```

Comment:
I know you don't like early returns… but this one _is_ an early return, so you might as well do it earlier. :innocent: 
```rust
let panorama = match state.confirmation_panorama(self.vidx, vhash) {
    Some(panorama) => panorama,
    None => return effects,
};
```

Reply:
Done in [`4052f72` (#572)](https://github.com/CasperLabs/casper-node/pull/572/commits/4052f72d1f3a987729feb79886716d40dc756ff6).